lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -3072,6 +3072,30 @@ void swift::checkTopLevelActorIsolation(TopLevelCodeDecl *decl) {
   decl->getBody()->walk(checker);
 }
 
+static bool checkDuplicateIsolatedParameter(AbstractFunctionDecl *decl) {
+  // If we have already seen at least one 'isolated' parameter,
+  // and we see another one, this is invalid as only one 'isolated' param
+  // per function is allowed.
+  // See:
+  // https://github.com/apple/swift-evolution/blob/main/proposals/0313-actor-isolation-control.md#multiple-isolated-parameters
+  ParamDecl* previousIsolatedParam = nullptr;
+  for (size_t i = 0; i < decl->getParameters()->size(); ++i) {
+    auto param = decl->getParameters()->get(i);
+    if (param->isIsolated()) {
+      if (auto previous = previousIsolatedParam) {
+        decl->diagnose(diag::isolated_parameter_second_isolated_param,
+                       previous->getParameterName(),
+                       param->getParameterName());
+        return true;
```

Comment:
I have a couple of optional suggestions:

1) It could be worth highlighting the extra parameters (via `.highlight`/`.highlightChars`). 
2) It could be worth emitting a note for each extra parameter (it seems to me that this diagnostic only works if you have 2-3 isolated parameters but if you have 4 then we will not mention one of those extra parameters).
3) It could be worth emitting a fix-it that removes `isolated` from the extra parameters.

Reply:
I took approach (2) in https://github.com/apple/swift/pull/61862